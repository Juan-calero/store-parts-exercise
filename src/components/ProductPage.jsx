import React from "react";
import { string, shape } from "prop-types";
import { productSection, btns } from "./productPage.module.scss";
import { Link } from "react-router-dom";

const ProductPage = ({ location: { name, price, type } }) => {
	return (
		<section className={productSection}>
			<ul>
				<li>
					<h2>Name:</h2>
					<p> {name}</p>
				</li>
				<li>
					<h2>Type:</h2>
					<p> {type}</p>
				</li>
				<li>
					<h2>Price:</h2>
					<p> {price}</p>
				</li>
			</ul>
			<div className={btns}>
				<Link to="/">Go Back</Link>
			</div>
		</section>
	);
};

ProductPage.propTypes = {
	location: shape({
		name: string.isRequired,
		type: string.isRequired,
		price: string.isRequired,
	}),
};

export default ProductPage;
