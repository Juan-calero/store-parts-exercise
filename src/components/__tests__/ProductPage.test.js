import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import ProductPage from "../ProductPage";

test("Should match Snapshot", () => {
	const product = {
		location: { name: "Mouse 1", price: "112.00$", type: "Mouse" },
	};
	render(
		<Router>
			<ProductPage {...product} />
		</Router>
	);
	const tree = renderer
		.create(
			<Router>
				<ProductPage {...product} />
			</Router>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
