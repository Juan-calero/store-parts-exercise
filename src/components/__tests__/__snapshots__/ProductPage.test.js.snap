// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match Snapshot 1`] = `
<section
  className="productSection"
>
  <ul>
    <li>
      <h2>
        Name:
      </h2>
      <p>
         
        Mouse 1
      </p>
    </li>
    <li>
      <h2>
        Type:
      </h2>
      <p>
         
        Mouse
      </p>
    </li>
    <li>
      <h2>
        Price:
      </h2>
      <p>
         
        112.00$
      </p>
    </li>
  </ul>
  <div
    className="btns"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Go Back
    </a>
  </div>
</section>
`;
